### PR TITLE
Add IBAN analyzer

### PR DESCRIPTION
**Pull request by Double-A-92**
As requested in issue #35
Not sure if this needs to be attached somehow, or it's automatically loaded?

### CODE REVIEW

File: analyzers/ibananalyzer.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+import re
+
+from .regexanalyzer import RegexAnalyzer
+
+
+class IBANAnalyzer(RegexAnalyzer):
+    def __init__(self, action, validate=False):
+        # Regex adapted from https://stackoverflow.com/a/44657292/3621482
```
Comment (d-Rickyy-b) [inline analyzers/ibananalyzer.py]: Adding the source of the regex is 👌 .

File: analyzers/ibananalyzer.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        """Checks if the given string could be a valid IBAN. Adapted from https://rosettacode.org/wiki/IBAN#Python."""
+
+        # Ensure upper alphanumeric input
+        potential_iban = potential_iban.replace(' ', '').replace('\t', '').replace('-', '')
+        if not re.match(r"^[\dA-Z]+$", potential_iban):
+            return False
+
+        # Validate if length matches the expected length based on the country code
+        country_code = potential_iban[:2]
+        if len(potential_iban) != self._iban_length_by_country.get(country_code, None):
```
Comment (d-Rickyy-b) [inline analyzers/ibananalyzer.py]: Nice additional check

File: analyzers/ibananalyzer.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            return False
+
+        # Official validation ( https://en.wikipedia.org/wiki/International_Bank_Account_Number#Validating_the_IBAN )
+        rearranged_iban = potential_iban[4:] + potential_iban[:4]
+        integer_iban = int(''.join(str(int(ch, 36)) for ch in rearranged_iban))  # BASE 36: 0..9,A..Z -> 0..35
+        return integer_iban % 97 == 1
+
+    _iban_length_by_country = dict(
```
Comment (d-Rickyy-b) [inline analyzers/ibananalyzer.py]: Wow, that's a lot of countries / different lenghts. I thought they are a bit more ... unified.

File: analyzers/__init__.py [hunk 4]
```diff
@@ -11,6 +11,7 @@
 from .regexanalyzer import RegexAnalyzer
 from .urlanalyzer import URLAnalyzer
 from .wordanalyzer import WordAnalyzer
+from .ibananalyzer import IBANAnalyzer
```
Comment (d-Rickyy-b) [inline analyzers/__init__.py]: Perfect :)

**Comment by Double-A-92** [comment]
Ok Added it to the init. The testing was actually really useful while developing, since those regex and checks can be fiddly and break easy with small changes. :)

Btw I noticed that the CreditCardAnalyzer might not actually work correctly... Or only if the CC numbers are nicely line by line without other text.

**Pull request merged**